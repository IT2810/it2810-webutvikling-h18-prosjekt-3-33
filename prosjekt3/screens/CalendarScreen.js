import React from 'react';
import {
	View,
  	Text,
  	Button, ScrollView,
  	AsyncStorage, StyleSheet
} from 'react-native';
import * as Storage from '../components/Storage.js'
import { Calendar, CalendarList, Agenda, calendarTheme} from 'react-native-calendars';

export default class CalendarScreen extends React.Component {
	constructor(props){
        super(props);
        this.state = {
        }
      }

	static navigationOptions = {
	    title: 'Calendar',
	};


  	render() {

		return (
			<ScrollView>
			<Agenda
			items ={
				{'2018-10-01': [{text: 'Morning appointment'},{text: 'Midday appointment'},{text: 'Afternoon appointment'}],
				 '2018-10-02': [{text: 'item 2 - any js object'}],
				 '2018-10-03': [],
				 '2018-10-04': [{text: 'item 3 - any js object'},{text: 'any js object'}],
				}}
			
			renderItem={(item,firstItemInDay) => {return (<View>
														<Text style = {styles.item}>{firstItemInDay}</Text>
														<Text style = {styles.item}>{item.key}</Text>
														</View>)}}
			renderEmptyDate={() => {return (<View><Text></Text></View>);}}
			rowHasChanged={(r1, r2) => {return r1.text !== r2.text}}
			/>
			</ScrollView>
	    )
  	}
}


const styles = StyleSheet.create({
    item: {
      padding: 10,
      fontSize: 18,
      height: 44,
    }
  }); 


