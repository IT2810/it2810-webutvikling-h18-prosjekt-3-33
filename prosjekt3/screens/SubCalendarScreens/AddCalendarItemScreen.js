import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet, Button, TextInput} from 'react-native';
import DatePicker from 'react-native-datepicker'
import * as Storage from '../../components/Storage';


export default class AddCalendarItem extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            date:"",
            startTime:'10:00',
            endTime: '12:00',
            text: '',

        }
      }
  static navigationOptions = {
    title: 'Create task ',
  };


  componentDidMount(){ //When components mounts, setState date equal to todays actual date using javascript date Object
      const date = new Date()
      d = date.toISOString().split('T'); //Splitting object at T to an array 
      this.setState({date: d[0]}) //Index 0 of array is todays date

  }

  render() {
  return(
      <View style={styles.container}>
      <Text>Select day</Text>
      <DatePicker
        style={{width: 200}}
        date={this.state.date}
        mode="date"
        placeholder="select date"
        format="YYYY-MM-DD"
        minDate="2017-01-01"
        maxDate="2019-12-30"
        confirmBtnText="Confirm"
        cancelBtnText="Cancel"
        showIcon={false}
        onDateChange={(date) => {this.setState({date: date})}}
      />
      <Text>Select start time</Text>

      <DatePicker
          style={{width: 200}}
          date={this.state.startTime}
          mode="time"
          format="HH:mm"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          minuteInterval={10}
          Is24Hour={true}
          showIcon={false}
          onDateChange={(time) => {
              this.setState({startTime: time});
            }}
        />
        <Text>Select end time</Text>
        <DatePicker
          style={{width: 200}}
          date={this.state.endTime}
          mode="time"
          format="HH:mm"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          minuteInterval={10}
          Is24Hour={true}
          showIcon={false}
          onDateChange={(time) => {
                this.setState({endTime: time});
            }}
        />

        <TextInput
        style={styles.input}
        value={this.state.text}
        multiline={true}
        maxLength={140}
        onChangeText={text => this.setState({text})}
        keyboardType="default"
        placeholder="information about your schedule"
        />
            {/*Button has an onPress function that sends us back to CalendarScreen with correct date item
            such that it can be displayed in our <Agenda /> */}
        <Button title="Add Task" onPress={()=>{
            alert("Added")
            const date = this.state.date
            const startTime = this.state.startTime
            const endTime = this.state.endTime
            const text = this.state.text
            const data = {
                date: date,
                startTime: startTime,
                endTime: endTime,
                text: text
            }
            this.props.navigation.state.params.onAddItem(data)
            this.props.navigation.navigate('Calendar')

        }}></Button>

        </View>
    )

  }
}


const styles = StyleSheet.create({
    item: {
      padding: 10,
      fontSize: 18,
      height: 44,
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'white'
      },
      input: {
        margin: 20,
        marginBottom: 0,
        height: 34,
        width: '80%',
        height: 200,
        paddingHorizontal: 10,
        borderRadius: 4,
        borderColor: '#ccc',
        borderWidth: 1,
        fontSize: 16,

      },
  }); 