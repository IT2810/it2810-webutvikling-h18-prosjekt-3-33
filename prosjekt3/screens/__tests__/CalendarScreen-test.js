import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CalendarScreen from '../CalendarScreen';
import AddCalendarItemScreen from '../SubCalendarScreens/AddCalendarItemScreen';
import {Agenda} from 'react-native-calendars'
import {DatePicker} from 'react-native-datepicker';


jest.mock('react-native-datepicker');
jest.mock('react-native-calendars');

describe('testing CalendarScreen', () =>{

    it('Renders AddCalenderItemScreen correctly', () => {
        const tree = renderer.create(<AddCalendarItemScreen />).toJSON();
        expect(tree).toMatchSnapshot();

    })

    it('Handle TextInput events in AddCalenderItemScreen', () => {
        const addCalendarItemScreen = renderer.create(<AddCalendarItemScreen />).getInstance();

        let testState = {
            date: '2018-10-05',
            startTime:'20:00',
            endTime: '23:00',
            text: 'Test text',

        }
        addCalendarItemScreen.handleDateChange(testState.date)
        addCalendarItemScreen.handleTextChange(testState.text)
        addCalendarItemScreen.handleStartTimeChange(testState.startTime)
        addCalendarItemScreen.handleEndTimeChange(testState.endTime)

        expect(addCalendarItemScreen.state).toEqual(testState)


    })

    it('Renders CalendarScreen correctly', () => {
        const navigation = {
            setParams: jest.fn(),
        }
        const tree = renderer.create(<CalendarScreen navigation={navigation}/>).toJSON();
        expect(tree).toMatchSnapshot();

    })

    it('onAddItem updates state correctly, CalendarScreen', () => {
        const navigation = {
            setParams: jest.fn(),
        }
        const Calendar = renderer.create(<CalendarScreen navigation={navigation}/>).getInstance();

        let testItem = {
            date: '2018-10-05',
            startTime:'20:00',
            endTime: '23:00',
            text: 'Test text',
        }

        let testState = {
            '2018-10-05': [
                {
                    name: '20:00-23:00',
                    text: 'Test text'
                }
            ]
        }
        Calendar.onAddItem(testItem)

        expect(Calendar.state.items).toEqual(testState)

    })

    it('onMountAddTasks updates state correctly, CalendarScreen', () => {
        const navigation = {
            setParams: jest.fn(),
        }
        const Calendar = renderer.create(<CalendarScreen navigation={navigation}/>).getInstance();

        let testItem = [{
            'key': '2018-10-05',
            'info': {
                    name: '20:00-23:00',
                    text: 'Test text'
                }
            }]

        let testState = {
            '2018-10-05': 
                {
                    name: '20:00-23:00',
                    text: 'Test text'
                }
        }

        Calendar.onMountAddTasks(testItem)
        expect(Calendar.state.items).toEqual(testState)

    })

    it('Calling onAddItem and checks if Snapshot matches, CalendarScreen', () => {
        const navigation = {
            setParams: jest.fn(),
        }
        const Calendar = renderer.create(<CalendarScreen navigation={navigation}/>)

        let testItem = {
            date: '2018-10-05',
            startTime:'20:00',
            endTime: '23:00',
            text: 'Test text',
        }

        Calendar.getInstance().onAddItem(testItem)
        expect(Calendar.toJSON()).toMatchSnapshot();

    })

})



