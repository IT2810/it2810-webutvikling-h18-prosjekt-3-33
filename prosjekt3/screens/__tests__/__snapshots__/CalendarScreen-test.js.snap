// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CalendarScreen Renders AddCalenderItemScreen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Select day
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Select start time
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Select end time
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    maxLength={140}
    multiline={true}
    onChangeText={[Function]}
    placeholder="information about your schedule"
    style={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "fontSize": 16,
        "height": 200,
        "margin": 20,
        "marginBottom": 0,
        "paddingHorizontal": 10,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Task
      </Text>
    </View>
  </View>
</View>
`;
